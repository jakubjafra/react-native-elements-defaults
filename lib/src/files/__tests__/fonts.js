const testHelperUtils = require('../../testHelperUtils');

const utils = require('../../utils');
const Overwrite = utils.Overwrite;

const name = '$fonts',
    __oldElement__ = require('react-native-elements/src/config/fonts'),
    __element__ = require('../fonts'),
    basicTheme = new Overwrite(
        name,
        Overwrite.TYPE.FILE,
        Overwrite.SCOPE.DEFAULTS_DEEP,
        {
            android: {
                regular: {
                    fontFamily: 'serif'
                }
            }
        }
    );

describe(name, () => {
    let OldElement, Element;

    beforeAll(() => {
        let defaultStyles = basicTheme;

        OldElement = __oldElement__.default;
        Element = __element__.compose(defaultStyles)(OldElement);
    });

    test(`config/${name} and ${name} should be different`, () => {
        expect(OldElement).not.toBe(Element);
    });

    test(`config/${name} are defined`, () => {
        expect(OldElement).toMatchSnapshot();
    });

    test(`${name} are defined`, () => {
        expect(Element).toMatchSnapshot();
    });

    test(`${name} contains overwriten property for primary color`, () => {
        expect(Element.android.regular.fontFamily).toEqual('serif');
    })
});
