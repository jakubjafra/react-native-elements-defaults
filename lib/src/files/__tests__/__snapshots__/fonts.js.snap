// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`$fonts $fonts are defined 1`] = `
Object {
  "android": Object {
    "black": Object {
      "fontFamily": "sans-serif",
      "fontWeight": "bold",
    },
    "bold": Object {
      "fontFamily": "sans-serif",
      "fontWeight": "bold",
    },
    "condensed": Object {
      "fontFamily": "sans-serif-condensed",
    },
    "condensed_light": Object {
      "fontFamily": "sans-serif-condensed",
      "fontWeight": "light",
    },
    "light": Object {
      "fontFamily": "sans-serif-light",
    },
    "medium": Object {
      "fontFamily": "sans-serif-medium",
    },
    "regular": Object {
      "fontFamily": "serif",
    },
    "thin": Object {
      "fontFamily": "sans-serif-thin",
    },
  },
  "ios": Object {},
}
`;

exports[`$fonts config/$fonts are defined 1`] = `
Object {
  "android": Object {
    "black": Object {
      "fontFamily": "sans-serif",
      "fontWeight": "bold",
    },
    "bold": Object {
      "fontFamily": "sans-serif",
      "fontWeight": "bold",
    },
    "condensed": Object {
      "fontFamily": "sans-serif-condensed",
    },
    "condensed_light": Object {
      "fontFamily": "sans-serif-condensed",
      "fontWeight": "light",
    },
    "light": Object {
      "fontFamily": "sans-serif-light",
    },
    "medium": Object {
      "fontFamily": "sans-serif-medium",
    },
    "regular": Object {
      "fontFamily": "sans-serif",
    },
    "thin": Object {
      "fontFamily": "sans-serif-thin",
    },
  },
  "ios": Object {},
}
`;
